### ISSUE TITLE
Assertion failed: mode == "constant" && value == 0.f && "This version of TensorRT only supports constant 0 padding!"

### ISSUE THREAD

**Issue by TomatoBoy90** [NVIDIA/TensorRT#1019]
Input filename:   XXX.onnx
ONNX IR version:  0.0.6
Opset version:    11
Producer name:    pytorch
Producer version: 1.7
Domain:           
Model version:    0
Doc string:
 
[01/18/2021-12:40:00] [W] [TRT] onnx2trt_utils.cpp:220: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
ERROR: builtin_op_importers.cpp:2191 In function importPad:
[8] Assertion failed: mode == "constant" && value == 0.f && "This version of TensorRT only supports constant 0 padding!"
[01/18/2021-12:40:00] [E] Failed to parse onnx file
[01/18/2021-12:40:00] [E] Parsing model failed
[01/18/2021-12:40:00] [E] Engine creation failed
[01/18/2021-12:40:00] [E] Engine set up failed

## Environment

**TensorRT Version**: TensorRT-7.1.3.4
**GPU Type**: v100
**Nvidia Driver Version**: 450.51.05
**CUDA Version**: 11.0
**CUDNN Version**: 8.0
**Operating System + Version**: centos7
**Python Version (if applicable)**: 3.7
**TensorFlow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 1.7
**Baremetal or Container (if container which image + tag)**:

**Comment by TomatoBoy90** [comment]
Now the problem is that pading is not supported.How can I solve it？

**Comment by TomatoBoy90** [comment]
when I use "TensorRT-7.1.3.4/bin/trtexec XXXX.onnx XXXX.trt",the above bug appears

**Comment by ttyio** [comment]
Sorry @TomatoBoy90 , current we only support constant 0 padding in op `Pad`.
Adding @kevinch-nv , do you have any workaround? thanks!

**Comment by ttyio** [comment]
@TomatoBoy90 , now we are working on Padding enhancement, it's helpful if you could share us with your model, thanks!

**Comment by urvangjoshi** [comment]
@ttyio : I'm actually having the same issue, and I can share my sample model.
Could you pls reopen the bug?
Thanks!

**Comment by ttyio** [comment]
Thanks @urvangjoshi, could you share the model?

**Comment by urvangjoshi** [comment]
Here you go (zipped file, pls unzip to get the .onnx file)
[sample_with_mirror_pad.zip](https://github.com/NVIDIA/TensorRT/files/6921157/sample_with_mirror_pad.zip)

**Comment by ttyio** [comment]
@urvangjoshi , this will be fixed in next release ( around 2 months from now), thanks

**Comment by urvangjoshi** [comment]
Good to know! Thanks.

**Comment by dmagee** [comment]
Glad to hear it will be included soon. If you want another test network there's one here:

https://drive.google.com/file/d/1GW3_gsQrNiTlHeE1Bvsqp5MgRN5fSXYN/view?usp=sharing

It was generated by saving the output from this:

https://github.com/LynnHo/CycleGAN-Tensorflow-2

 as .h5 and then using tf2onnx

Thanks

Derek

**Comment by devalexqt** [comment]
Is it fixed in TRT 8.2 ?

**Comment by ttyio** [comment]
Yes @devalexqt , please try 8.2EA in https://developer.nvidia.com/nvidia-tensorrt-8x-download, thanks

**Comment by p1x31** [comment]
still having this issue with:
TensorRT-8.2.3.0
onnx 1.10.2
opset_version=14
model: https://drive.google.com/file/d/1rv-hQwhLS5OEXx8hGLi-0HUprGea7yZb/view?usp=sharing

```
While parsing node number 0:
builtin_op_importers.cpp:3004 In function importPad:
[8] Assertion failed: mode == "constant" && value == 0.f && "This version of TensorRT only supports constant 0 padding!"
```

**Comment by dennex** [comment]
Are there any workarounds if we are using TensorRT version 8.0.16?
Our C++ code, CUDA, CuDNN are all setup with TensorRT 8.0.16 and it would break a lot of projects to upgrade to 8.2

**Comment by ttyio** [comment]
@dennex , I do not know any easy workarounds for this, we add cuda kernel to implement this, maybe you need write your own plugin.

**Comment by nvpohanh** [comment]
@p1x31 Could you try TRT 8.4.1 (8.4GA) release? @ttyio I think Pad with Reflect mode is now supported, right?

**Comment by nvpohanh** [comment]
Closing due to >14 days without activity. Please feel free to reopen if the issue still exists in TRT 8.4. Thanks

**Comment by dennex** [comment]
> Are there any workarounds if we are using TensorRT version 8.0.16? Our C++ code, CUDA, CuDNN are all setup with TensorRT 8.0.16 and it would break a lot of projects to upgrade to 8.2

We've actually upgraded to 8.4 and it works well.
Some other layer types didnt exist/were not recognized, which we found the ONNX-simplifier to be a useful post-processing step to our ONNX.

**Comment by Shaw-Way** [comment]
> @p1x31 Could you try TRT 8.4.1 (8.4GA) release? @ttyio I think Pad with Reflect mode is now supported, right?

Hi, how to set reflect pad mode with trt? I couldn't find the api

**Comment by nvpohanh** [comment]
ISliceLayer supports padding with reflect mode: https://docs.nvidia.com/deeplearning/tensorrt/api/python_api/infer/Graph/Layers.html#islicelayer

**Comment by Shaw-Way** [comment]
> ISliceLayer supports padding with reflect mode: https://docs.nvidia.com/deeplearning/tensorrt/api/python_api/infer/Graph/Layers.html#islicelayer

Thanks for answering.
